### ISSUE TITLE
Halifax Mobile Banking

### ISSUE THREAD

**Issue by brxken128** [PrivSec-dev/banking-apps-compat-report#309]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

Halifax Mobile Banking

### Link to app

https://play.google.com/store/apps/details?id=com.grppl.android.shell.halifax

### App version

123.04

### Country of the app

United Kingdom

### Build Number

UP1A.231005.007.2023101300

### Device list

Pixel 6

### Profile app tested in

Secondary profile(s)

### Google Play installed

Installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [ ] Enabled
- [X] Disabled

### Exploit protection compatibility mode

- [ ] Enabled
- [X] Disabled

### Stock OS compatibility

- [X] Works
- [ ] Does not work
- [ ] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [X] N/A (Not supported by app)
- [ ] Not tested

### Description of the app's functionality

The app works great, although signing in is a pain if you use a randomly generated password as switching between a password manager and the app resets all of the details - I think this is just the app, and no fault of gOS as I've experienced it on my stock P8 Pro (will be installing gOS on that shortly, I'm pretty certain it'll still work fine though).

Of course, there's no working Google Wallet integration.

### Are there any extra notes you think users should know about?

As with my last submission, I'm not 100% on the build number but it was the [2023101300](https://grapheneos.org/releases#2023101300) release.

**Comment by brxken128** [comment]
By the way @akc3n, the issue form requests that you add the appropriate country label but only people with triage permissions can do that (I'm not sure if it's geared towards the general population or for internal members though).

**Comment by akc3n** [comment]
@brxken128 thanks for reminding me of that. I'll update the instructions on the form.

**Comment by brxken128** [comment]
Are reports from the same bank, but different devices supported? I just flashed my Pixel 8 pro and use 4/5 UK banking apps.

**Comment by akc3n** [comment]
@brxken128 Yes. Issue authors may edit their own banking app report, add additional devices they've tested on, and update the notes field to reflect changes, workarounds, what devices it applies to, version of app, and os build numbers.
> Are reports from the same bank, but different devices supported?

For non-authors of a submitted banking app report it might be best to add a comment with those details. That way users and contributors whom subscribe to specific app reports can be notified of changes. Making compatibility with GrapheneOS easier to track.
> I just flashed my Pixel 8 pro and use 4/5 UK banking apps.

I should make a snippet template for this sort of things to make things easier when contributors wish to update others banking app reports in the comments.

**Comment by spring-onion** [comment]
As seen in #346 the app now denies functionality, probably due to play integrity. Please check out [this](https://github.com/PrivSec-dev/banking-apps-compat-report/issues/342) update to find out what's being done to combat this problem and what you can do to help.

**Comment by spring-onion** [comment]
Thanks to the recent improvements in avoiding detection by third party sdks and also working around buggy play integrity implementations, Halifax appears to be working again.

**Comment by doogstar** [comment]
I'm still having problems with Halifax (and mirrored in Lloyds app) on their version 156.03, currently on GOS 2025030800.

After switching on Exploit protection compatibility mode and clearing storage, the first launch took me through the enrolment process, called me with a 4 digit code, and let me in to my accounts. 
Subsequent launches now give:
"Sorry, the app won't start. Sorry, your device has been de-registered for security reasons. You'll need to re-register your device to use the app again. Please call us on +44xxxxxx for more information (8)"

**Comment by RMED24** [comment]
Can confirm that I'm suffering from the same issue as @doogstar 

Deregistered device for security after it was all set up

**Comment by spring-onion** [comment]
@RMED24 

Marking incompatible again. Are you notified of play integrity usage at any point? Keep me updated if anything changes.

**Comment by doogstar** [comment]
Been through the process again, verified pin through the automated call then same message. No play integrity notification shown. Have written down number and will plan to call them, especially as my old backup phone has started receiving notifications that they're going to drop Android 7 support from the app shortly, so that won't be an option anymore. Depending on what they say will probably start a formal complaint.

**Comment by RMED24** [comment]
Same hear. No play integrity and it let's me get through the whole process again before saying it is getting deregistered for security. As far as I remember, this never happened on my old phone with stock ROM.

If there's any other information you need, let me know. I haven't collected much else though
